wemake_python_styleguide/visitors/tokenize/primitives.py

```diff
@@ -30,12 +31,31 @@ class WrongNumberTokenVisitor(BaseTokenVisitor):
         'X', 'O', 'B', 'E',
     ))
 
+    _bad_number_prefixes = ('0b0', '0x0', '0o0', 'e0')
+
     def _check_underscored_number(self, token: tokenize.TokenInfo) -> None:
         if '_' in token.string:
             self.add_violation(
                 UnderscoredNumberViolation(token, text=token.string),
             )
 
+    def _check_meaningless_zeros(self, token: tokenize.TokenInfo) -> None:
+        token_string = token.string.lower()
+        token_start_pos = token_string.find('e')
+        token_start_pos = token_start_pos if token_start_pos >= 0 else 0
+        for prefix in self._bad_number_prefixes:
+            if token_string[token_start_pos:].startswith(prefix):
+                count_digits = len(token_string[2:])
+                if prefix != 'e0' and count_digits & (count_digits - 1) == 0:
```

Comment:
Can you please clarify what `count_digits & (count_digits - 1)` is?

Reply:
It is checking that literals have 2**n digits
For example:
4 = 0b0100
&
(4-1) = 0b0011
=
0

Reply:
@SizovIgor ping